### ISSUE TITLE
Apple’s version 1 kern subtable is not supported

### ISSUE THREAD

**Issue by khaledhosny** [foliojs/fontkit#127]
I was testing some of the Arabic AAT fonts shipped with macOS and noticed some glyph positioning issues, looking into the code the `AATLayoutEngine` class does not have a `position` method, does this mean AAT glyph positions are not currently supported? If so, any pointers where to start if one wanted to work on it?

**Comment by khaledhosny** [comment]
After a bit of research I found that glyph positioning in AAT is handled by `kern` table which is already supported by fontkit (including most Apple subtable formats), but the fonts I was testing has version 1 kern subtable which fontkit does not currently support.

**Comment by devongovett** [comment]
Yeah, state machine based kerning is not currently supported in either the `kern` or newer `kerx` table. It shouldn't be that difficult to implement though since we already have the shared AAT state machine implementation.

**Comment by devongovett** [comment]
I remembered doing some work on this at some point, and managed to find some code buried in an old git stash. I've committed it to the [aat-kern](https://github.com/devongovett/fontkit/compare/aat-kern) branch if you feel like taking a look at it. I don't really remember how far I got, but it seems like support for decoding format 1 kern subtables is there, along with the `kerx` table. Additionally, it looks like a rewrite of `KernProcessor` had been started, along with some AAT state machine processing.

I think my main problem was finding fonts to test with. Most of the AAT fonts that exist are built into macOS. If you know of any open source fonts that do state table based kerning, let me know.

Anyway, feel free to work from that branch if you feel like it. Again, I don't really remember how much is working, but the existing tests do pass at least so that's good. :smile:
